src/java/org/apache/cassandra/db/compression/CompressionDictionary.java

```diff
@@ -188,9 +220,10 @@ enum Kind
         // Order matters: the enum ordinal is serialized
         ZSTD
         {
-            public CompressionDictionary createDictionary(DictId dictId, byte[] dict)
+            @Override
+            public CompressionDictionary createDictionary(DictId dictId, byte[] dict, int checksum)
             {
-                return new ZstdCompressionDictionary(dictId, dict);
+                return new ZstdCompressionDictionary(dictId, dict, checksum);
```

Comment:
This seems way more comfortable to do. If a reviewer looks into where checksum is worked with, we have it every time, no reason to not use that so we do not need to compute it.